trunner/test/test_harness.py

```diff
@@ -0,0 +1,240 @@
+#
+# Phoenix-RTOS test runner
+#
+# Tests for Mbedtls Test Suite harness
+#
+# Copyright 2022 Phoenix Systems
+# Authors: Damian Loewnau
+#
+
+import os
+import threading
+
+import pytest
+import pexpect.fdpexpect
+
+from trunner.harnesses.mbedtls import MbedtlsTestHarness
+from trunner.harnesses.common import TestResult
+from trunner.testcase import TestCase
+from trunner.config import ALL_TARGETS
+
+TestResult.__test__ = False
+
+
+def get_prepared_msg(testcases_msg):
+    testcases_msg = testcases_msg.replace('  ', '\t\t')
+    testcases_msg = testcases_msg.replace('\r', '')
+
+    return testcases_msg
+
+
+def assert_results(result, expected):
+    for i in range(len(expected)):
+        assert result[i].status == expected[i].status
+        assert result[i].name == expected[i].name
+        assert result[i].msg == expected[i].msg
+
+
+def write_output(output, fifo, lock):
+    with open(fifo, 'w') as f:
+        f.write(output)
+        f.flush()
+
+        # We've written what we wanted to
+        # Keep open fifo to not trigger EOF, hanging on a lock
+        lock.acquire()
+
+
+class TestMbedtlsHarness:
+    """ Tests assert that Mbedtls harness parses Mbedtls Test Suite output properly """
+
+    FIFO = "./test_harness.fifo"
+
+    @pytest.fixture
+    def testcase(self):
+        return TestCase(
+            name='xyz',
+            target=ALL_TARGETS[0],
+            timeout=1,
+            exec_cmd=['xyz'],
+            psh=False
+        )
+
+    @pytest.fixture(scope="class")
+    def fifo(self):
+        os.mkfifo(self.FIFO)
+        yield
+        os.unlink(self.FIFO)
+
+    def run_harness(self, writer_thr, assert_wrong=False, testcase=None):
+        lock = threading.Lock()
+        lock.acquire()
+
+        thread = threading.Thread(target=writer_thr, args=(lock,))
+        thread.start()
+        fifo_r = open(self.FIFO, 'r')
+
+        if not assert_wrong:
+            proc = pexpect.fdpexpect.fdspawn(fifo_r, encoding='ascii', timeout=5)
+            result = MbedtlsTestHarness.harness(proc)
+        else:
+            proc = pexpect.fdpexpect.fdspawn(fifo_r, encoding='ascii', timeout=1)
+            testcase.harness = MbedtlsTestHarness.harness
+            testcase.handle(proc)
```

Comment:
```suggestion
        proc = pexpect.fdpexpect.fdspawn(fifo_r, encoding='ascii', timeout=5)
        try:
            result = MbedtlsTestHarness.harness(proc)
        finally:
```
We misunderstood here. We would like to check if `MbetlsTestHarness.harness` throws exception - now we test if `TestCase` handles exception properly.